lib/libc/minimal/source/stdout/prf.c

```diff
@@ -461,16 +461,14 @@ int z_prf(int (*func)(), void *dest, const char *format, va_list vargs)
 	int width;
 	VALTYPE val;
 
+#define PUTC(c)	do { if ((*func)(c, dest) == EOF) return EOF; } while (0)
```

Comment:
Would you mind change `while (0)` to `while (false)` ? MISRA-C expects boolean operands in boolean expressions.